feat(core): reworked the walledgarden layout

jdalsem opened the pull request:
Most notable changes
- different look and feel
- dropped the javascript form loading behaviour (including the helper views for ajax loaded forms)
- dropped the walledgarden rounded corners images
- reduced the need for custom css in themes related to walled garden
- moved walled garden css loading to page shell (so no need to load it whereever you draw the walled garden)
- simplyfied and standardized usage of walled_garden pageshell/layout/module

Background Image used is CC0 Public Domain https://pixabay.com/en/lamp-jetty-wooden-pier-banister-677698/

Themes can easily replace the background image by adding the view /walled_garden.jpg

![walled garden](https://cloud.githubusercontent.com/assets/861833/19108166/f1989df2-8aef-11e6-9be9-0f12b93c68c2.png)

fixes #10305

jdalsem committed:
feat(core): reworked the walledgarden layout

mrclay commented on views/default/page/walled_garden.php:
```diff
@@ -9,29 +9,45 @@
  * @uses $vars['sysmessages'] A 2d array of various message registers, passed from system_messages()
  */
 
```
I wonder if we render some other view to reduce duplication here. Maybe not.

hypeJunction replied:
I think using the default shell should be fine

jdalsem replied:
don't think so... walled garden shell is a lot different then default shell

hypeJunction commented on mod/externalpages/views/default/expages/wrapper.php:
```diff
@@ -5,12 +5,17 @@
+	return;
+}
 
 echo '<div class="mtm">';
```
Make this a clearfix

hypeJunction commented on mod/externalpages/start.php:
```diff
@@ -67,19 +67,18 @@ function expages_page_handler($page, $handler) {
+	$description = $object ? $object[0]->description : elgg_echo('expages:notset');
+	
+	$content = elgg_view('expages/wrapper', [
+		'content' => elgg_view('output/longtext', ['value' => $description]),
```
Could you break this down into multiple lines, please

hypeJunction commented on mod/externalpages/start.php:
```diff
@@ -94,8 +93,7 @@ function expages_page_handler($page, $handler) {
 		$body = elgg_view_layout('one_column', array('title' => $title, 'content' => $content));
```
```
$layout = 'default';
if (elgg_get_config('walled_garden') && !elgg_is_logged_in()) {
   $layout = 'walled_garden';
}
$body = elgg_view_layout($layout, [
   'title' => $title,
   'content' => $content,
   'sidebar' => false, // or?
]);
```

hypeJunction commented on views/default/forms/user/changepassword.php:
```diff
@@ -3,39 +3,27 @@
  */
 
-$text = elgg_autop(elgg_echo('user:changepassword:change_password_confirm'));
+echo elgg_view('output/longtext', ['value' => elgg_echo('user:changepassword:change_password_confirm')]);
```
Break down into multiple lines

hypeJunction commented on views/default/forms/user/changepassword.php:
```diff
@@ -3,39 +3,27 @@
-));
-
-$u = elgg_view('input/hidden', array(
+echo elgg_view_input('hidden', [
```
Update to `elgg_view_filed()` unless it hasn't been yet

jdalsem replied:
was not available in master at that time...

hypeJunction replied:
I am going to merge up now

hypeJunction replied:
All branches are up to date now.

hypeJunction commented on views/default/forms/user/changepassword.php:
```diff
@@ -3,39 +3,27 @@
+	'name' => 'password2',
+]);
 
+$footer = elgg_view('input/submit', ['value' => elgg_echo('changepassword')]);
```
`elgg_view_field()`

hypeJunction commented on views/default/forms/user/requestnewpassword.php:
```diff
@@ -5,20 +5,16 @@
+
+echo elgg_view('input/captcha');
+
+$footer = elgg_view('input/submit', array('value' => elgg_echo('request')));
```
`elgg_view_field()`

hypeJunction commented on views/default/page/layouts/walled_garden.php:
```diff
@@ -2,15 +2,11 @@
+$title = elgg_extract('title', $vars);
+$body = elgg_extract('content', $vars);
+
+echo elgg_view_module('walled_garden', $title, $body);
```
Is this needed? We are passing `title` to layout view

hypeJunction commented on views/default/resources/account/change_password.php:
```diff
@@ -32,7 +32,6 @@
 
 if (elgg_get_config('walled_garden')) {
-	elgg_load_css('elgg.walled_garden');
 	$body = elgg_view_layout('walled_garden', array('content' => $content));
```
I think we should just pass the title into the layout and get rid of walled garden module wrapper.

jdalsem replied:
so passing title to layout would draw a title (just as the regular shell does) and completely drop the walled garden module as it is just normal content? all up for it 👍

hypeJunction commented on views/default/resources/account/forgotten_password.php:
```diff
@@ -1,29 +1,25 @@
 
-$title = elgg_echo("user:password:lost");
+$title = elgg_echo('user:password:lost');
+$content = elgg_view_form('user/requestnewpassword', ['class' => 'elgg-form-account']);
```
Pass the `$vars` to the form

hypeJunction commented on views/default/walled_garden.css.php:
```diff
@@ -2,82 +2,65 @@
-.elgg-walledgarden-single > .elgg-foot {
-	background: url(<?= elgg_get_simplecache_url("walled_garden/one_column_bottom.png"); ?>) no-repeat left top;
+body {
+	background-image: url('<?php echo elgg_get_simplecache_url('walled_garden.jpg'); ?>');
```
We don't really need to use php for this. 
`background-image: url(walled_garden.jpg);`

hypeJunction replied:
I want to eventually make these proper .css files - there is not reason for all the hacky `<style>` declarations.

jdalsem replied:
this php was the only thing not making them proper css files, so if i change the link, i will also change the file extension

hypeJunction replied:
👏

hypeJunction commented on views/default/walled_garden.css.php:
```diff
@@ -2,82 +2,65 @@
+.elgg-page-walled-garden {
+	position: relative;
+	right: 0;
+	width: 500px;
```
Do we need both `float` and `right`?

jdalsem replied:
nope

mrclay commented on views/default/walled_garden.css:
```diff
@@ -0,0 +1,58 @@
+body {
+	background-image: url('walled_garden.jpg');
```
This only allows replacement with other JPGs. #8721 Not a blocking issue though.

jdalsem commented:
Addressed all comments; most notable changes:
- using default layout, no more walled garden layout, only walledgarden shell
- no more walled garden modules
- using elgg_view_field for forms

as far as i can see this is ready for a merge...

mrclay commented:
On register page, short viewports push the .elgg-page-footer menu into the register button.
